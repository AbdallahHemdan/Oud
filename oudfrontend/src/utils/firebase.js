import firebase from "firebase";

const firebasConfig = {
  apiKey: "AIzaSyBZVUambAFEB-ywzAZTAPP-tSkbYXSEWc4",
  authDomain: "oud-notifications.firebaseapp.com",
  databaseURL: "https://oud-notifications.firebaseio.com",
  projectId: "oud-notifications",
  storageBucket: "oud-notifications.appspot.com",
  messagingSenderId: "681943816425",
  appId: "1:681943816425:web:9149a8834b2ff0f4623878"
};
// Initialize Firebase
firebase.initializeApp(firebasConfig);

export default firebase;
