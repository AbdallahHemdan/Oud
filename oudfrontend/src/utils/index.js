import axios from 'axios'
import { base } from '../config/environment'
const findByTestAttr = (component, val) => {
    return component.find(`[data-testid="${val}"]`);
}

function addToLikedSongs(id) {
    axios.put(`${base}/me/tracks/${id}`)
        .then(function (response) {
            console.log(response);

        })
        .catch(function (error) {
            console.log(error);
        });
}
function removeLikedSong(id) {
    axios.delete(`${base}/me/tracks/${id}`)
        .then(function (response) {
            console.log(response);
        })
        .catch(function (error) {
            console.log(error);
        });
}
export { findByTestAttr, addToLikedSongs, removeLikedSong }
