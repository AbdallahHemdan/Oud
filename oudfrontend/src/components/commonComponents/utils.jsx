import axios from "axios";
import { base } from "../../config/environment"
import {config} from "../../utils/auth"
/**
 * @file this file contains three functions used in Playlist, Album, LikedSongs 
 **/
export function addSong(playlistId, track){
  axios
  .post(`${base}/playlists/${playlistId}/${track}`, config)
  .then((response) => {
  })
  .catch((error) => {
  });
}