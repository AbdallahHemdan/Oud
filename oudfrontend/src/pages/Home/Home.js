import React, { Component } from "react";
import "./Home.css";
import Sidebar from "../../components/Sidebar/Sidebar";
import Navbar from "../../components/Navbar/Navbar";
import MainContent from "../../components/MainContent/MainContent";
import { base } from "./../../config/environment";
import axios from "axios";
import LoadingSnipper from "./../../components/LoadingSnipper/LoadingSnipper";
import { config, isLoggedIn } from "./../../utils/auth";

/**
 * a string to store endpoint url of getting List of Categories
 *
 * @type {string}
 *
 */
let fetchCategoriesUrl = `${base}/browse/categories`;

/**
 * Component to render all the stuff in Home page
 *
 * @author Abdallah Hemdan
 *
 * @component
 *
 */

class Home extends Component {
  /**
   * @constructor

   * @public

   * @param {object} props - get props from higher components
   * @param {Array<object>} items - An array of all categories of the music to display it in home page
   * @param {number} limit - The maximum number of categories to get.
   * @param {number} offset - The index of the first categories to get.
   * @param {number} total - The total number of categories available to get.
   */
  constructor(props) {
    super(props);
    this.state = {
      /**
       * An array of all categories of the music to display it in home page
       *
       * @type {Array<object>}
       */
      items: [],

      /**
       * The maximum number of categories to get.
       *
       * @type {number}
       */
      limit: 0,

      /**
       * The index of the first categories to get.
       *
       * @type {number}
       */
      offset: 0,
      /**
       * The total number of categories available to get.
       *
       * @type {number}
       */
      total: 0,
      /**
       * Check if the data loaded from the backend or not
       */
      isLoading: true,
    };
  }

  /**
   * Take the data fetched from the api and store it in the local state
   *
   * @function
   *
   * @param {Array<object>} items - An array of all categories of the music to display it in home page
   * @param {number} limit - The index of the first categories to get.
   * @param {number} offset - The maximum number of categories to get.
   * @param {number} total - The total number of categories available to get.
   *
   * @return {void} returns nothing, it just store data in state
   */
  handleStoringCategories = ({ items, limit, offset, total }) => {
    this.setState({ items, limit, offset, total, isLoading: false });
  }

  /**
   * Fetching data of all categories immediately after the component has been mount to the DOM tree
   */
  componentDidMount() {
    axios
      .get(fetchCategoriesUrl) // get all categories
      .then((result) => {
        this.handleStoringCategories(result.data);
      })
      .catch((err) => {
        console.log(err);
      });
  }

  /**
   * @function
   * @name render
   * @description Render all Home components
   * @returns {JSX} Component for App
   */
  render() {
    return (
      <div>
        <Sidebar
          data-testid="sidebar"
        />
        <Navbar isLoggedIn={isLoggedIn()}
          data-testid="navbar"
        />
        {this.state.isLoading ? (
          <LoadingSnipper
            data-testid="loading"
          />
        ) : (
            <React.Fragment>
              <MainContent
                items={this.state.items}
                webPlayer={this.props.webPlayer}
                data-testid="music-content"
              />
            </React.Fragment>
          )}
      </div>
    );
  }
}

export default Home;
