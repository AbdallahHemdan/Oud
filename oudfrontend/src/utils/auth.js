import jwtDecode from "jwt-decode";
import { getRequest } from "./requester";
import { base } from "./../config/environment";

function getToken() {
  return localStorage.getItem("accessToken");
}

const config = {
  headers: { authorization: `Bearer ${getToken()}` }
};

function Auth() {
  try {
    if (!getToken()) return null;
    let decoded = jwtDecode(getToken());
    console.log(decoded.id);
    return decoded.id;
  } catch {
    return null;
  }
}
function isLoggedIn() {
  return getToken() ? true : false;
}

function isArtist() {
  return getRequest(`${base}/me`)
    .then(response => {
      return response.data.role === "artist";
    })
    .catch(error => {
      console.log(error);
    });
}
const _config = config;
export { _config as config };
export const token = getToken();
const _Auth = Auth;
export { _Auth as Auth };
const _isLoggedIn = isLoggedIn;
export { _isLoggedIn as isLoggedIn };
const _isArtist = isArtist;
export { _isArtist as isArtist };
