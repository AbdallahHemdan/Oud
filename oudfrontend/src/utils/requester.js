import axios from "axios";
import { isLoggedIn } from "./../utils/auth";
const config = {
  headers: {
    authorization: `Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjVlOTA3ZGIwYTA2NDVmNDU4MTYwNzYwNiIsImlhdCI6MTU5MTYyMTQxOSwiZXhwIjoxNTk0MjEzNDE5fQ.fj3N3Pc89Pf_xlt7fGmXw1SINTecUB4-y3pihAAPjC8`,
    "Access-Control-Allow-Origin": "*"
  }
};
const getRequest = endpoint => {
  return axios.get(endpoint, config);
};
/**
 * Axios DELETE request
 * @function
 * @param {string} endpoint the required endpoint to implement an action
 * @returns {object}
 */
const deleteRequest = endpoint => {
  return axios.delete(endpoint, config);
};
/**
 * Axios PUT request
 * @function
 * @param {string} endpoint the required endpoint to implement an action
 * @param {object} body the request body
 * @returns {object}
 */
const putRequest = (endpoint, body = {}) => {
  return axios.put(endpoint, body, config);
};
/**
 * Axios POST request
 * @function
 * @param {string} endpoint the required endpoint to implement an action
 * @param {object} body the request body
 * @returns {object}
 */
const postRequest = (endpoint, body = {}) => {
  return axios.post(endpoint, body, config);
};
/**
 * Axios PATCH request
 * @function
 * @param {string} endpoint the required endpoint to implement an action
 * @param {object} body the request body
 * @returns {object}
 */
const patchRequest = (endpoint, body = {}) => {
  return axios.patch(endpoint, body, config);
};

// const isArtist = () => {
//   if (isLoggedIn()) {
//   }
// };
export { getRequest, deleteRequest, putRequest, postRequest, patchRequest };
