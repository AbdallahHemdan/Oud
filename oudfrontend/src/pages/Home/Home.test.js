import React from "react";
import { shallow } from 'enzyme';
import { findByTestAttr } from "./../../utils/index"
import Home from './Home';


const setup = (props = {}) => {
  const component = shallow(
    <Home {...props} />
  )
  return component;
}

describe('Home Page Component', () => {
  let component;
  beforeEach(() => {
    component = setup();
  })

  it('should sidebar in right way', () => {
    const wrapper = findByTestAttr(component, "sidebar");
    expect(wrapper.length).toBe(1);
  });

  it('should navbar in right way', () => {
    const wrapper = findByTestAttr(component, "navbar");
    expect(wrapper.length).toBe(1);
  });

  it('Should render loading in case of loading data', () => {
    component.setState({ isLoading: true });
    const wrapper = findByTestAttr(component, "loading");
    expect(component.state().isLoading).toBeTruthy();
    expect(wrapper.length).toBe(1);
  });

  it('should music content in right way', () => {
    component.setState({ isLoading: false });
    const wrapper = findByTestAttr(component, "music-content");
    expect(component.state().isLoading).toBeFalsy();
    expect(wrapper.length).toBe(1);
  });

  describe('snapshot test of Home', () => {
    it('renders component correctly', () => {
      const wrapper = shallow(
        <Home />
      )
      expect(wrapper).toMatchSnapshot()
    });
  });
});