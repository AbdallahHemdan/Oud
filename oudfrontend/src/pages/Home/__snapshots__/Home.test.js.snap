// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page Component snapshot test of Home renders component correctly 1`] = `ShallowWrapper {}`;
